megamek/src/megamek/client/ui/swing/boardview/SensorRangeSprite.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright (c) 2023 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MegaMek.
+ *
+ * MegaMek is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MegaMek is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+package megamek.client.ui.swing.boardview;
+
+import java.awt.AlphaComposite;
+import java.awt.BasicStroke;
+import java.awt.Color;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.Image;
+import java.awt.Shape;
+import java.awt.Stroke;
+import java.awt.image.ImageObserver;
+import static megamek.client.ui.swing.boardview.HexDrawUtilities.*;
+
+import megamek.client.ui.swing.GUIPreferences;
+import megamek.client.ui.swing.util.UIUtil;
+import megamek.common.Coords;
+
+/**
+ * This sprite is used to paint the visual and sensor range
+ * 
+ * <BR><BR>Extends {@link MovementEnvelopeSprite}
+ * 
+ * @author Simon
+ */
+public class SensorRangeSprite extends MovementEnvelopeSprite {
```

Comment:
Since this extends MoveEnvSprite, are all the methods necessary? They look copied to me.

Reply:
it is a copy of FieldofFireSprite with a different  getColor function